internal/datanode/compaction/mix_compactor.go

```diff
@@ -329,17 +329,25 @@ func (t *mixCompactionTask) Compact() (*datapb.CompactionPlanResult, error) {
 		return nil, errors.New("illegal compaction plan")
 	}
 
-	allSorted := true
-	for _, segment := range t.plan.GetSegmentBinlogs() {
-		if !segment.GetIsSorted() {
-			allSorted = false
-			break
+	sortMergeAppicable := paramtable.Get().DataNodeCfg.UseMergeSort.GetAsBool()
+	if sortMergeAppicable {
+		if len(t.plan.GetSegmentBinlogs()) <= 0 {
+			sortMergeAppicable = false
+		}
```

Comment:
Empty inputs have been checked at L:329 and L:93, it'll report illegal compaction plan error.